src/ewah.zig

```diff
@@ -61,98 +64,208 @@ pub fn ewah(comptime Word: type) type {
             return @as(Word, @bitCast(mark));
         }
 
+        pub const Decoder = struct {
+            target_words: []Word,
+            target_index: usize = 0,
+            source_literal_words: usize = 0,
```

Comment:
One thing that gives me pause here is that there isn't any safety checks to ensure that the entire
thing was decoded. Basically, there's no difference between a Decoder which is driven to completion,
and a Decoder that missed some chunks (or processed a chunk twice).

Perhaps it makes sense to add expected total length and a `.finish()` method just for assertions?

Ah, curious how that problem doesn't exist for Encoder! Yeah, I think we should add `Decoder.done`,
so that the caller can `assert(decoder.done())`.